### PR TITLE
chore: doc specs location and scripts cleanup [skip-bc]

### PR DESCRIPTION
**Pull request by shortcuts**
## 🧭 What and Why

🎟 JIRA Ticket: https://algolia.atlassian.net/browse/DI-3319

### Changes included:

move the documentation specs to the docs directory, also cleanup the specs scripts a bit

### CODE REVIEW

File: scripts/specs/index.ts [hunk 1]
```diff
@@ -70,21 +65,22 @@ async function buildSpec({
...
     return;
   }
 
   // In case of lite we use a the `search` spec as a base because only its bundled form exists.
-  const specBase = isAlgoliasearch ? 'search' : spec;
-  const deps = isAlgoliasearch ? ['search', 'recommend'] : [spec];
+  const specBase = isLiteSpec ? 'search' : spec;
   const logSuffix = docs ? 'doc spec' : 'spec';
+  const basePath = docs ? 'docs/' : 'specs/';
+  const deps = isLiteSpec ? ['search', 'recommend'] : [spec];
```
Comment (millotp) [inline scripts/specs/index.ts:78]: `common` is not a deps anymore ?

File: scripts/specs/index.ts [hunk 2]
```diff
@@ -102,19 +98,19 @@ async function buildSpec({
...
+
   // Add the correct tags to be able to generate the proper client
-  if (!isAlgoliasearch) {
-    await transformBundle({
-      bundledPath: toAbsolutePath(bundledPath),
-      clientName: spec,
-      docs,
-    });
+  if (!isLiteSpec) {
+    docs ? await bundleSpecsForDoc(bundledPath, spec) : await bundleSpecsForClient(bundledPath, spec);
```
Comment (millotp) [inline scripts/specs/index.ts:113]:
```suggestion
    await (docs ? bundleSpecsForDoc(bundledPath, spec) : bundleSpecsForClient(bundledPath, spec));
```

File: scripts/specs/format.ts [hunk 3]
```diff
@@ -37,48 +36,26 @@ export async function lintCommon(useCache: boolean): Promise<void> {
...
-    await fsp.writeFile(toAbsolutePath(`docs/bundled/${clientName}-snippets.json`), snippets);
-  }
+  parseCodeSamples(JSON.parse(JSON.stringify(codeSamples)));
 
   for (const [pathKey, pathMethods] of Object.entries(bundledSpec.paths)) {
     for (const [method, specMethod] of Object.entries(pathMethods)) {
-      if (!docs) {
-        // In the main bundle we need to have only the clientName
-        // because open-api-generator will use this to determine the name of the client
-        specMethod.tags = [clientName];
```
Comment (millotp) [inline scripts/specs/format.ts:78]: this is not needed anymore ?

File: scripts/specs/format.ts [hunk 4]
```diff
@@ -142,12 +119,6 @@ export async function transformBundle({
           );
         }
 
-        if (alias && tag === alias) {
```
Comment (millotp) [inline scripts/specs/format.ts:145]: same

File: scripts/specs/format.ts (hunk 3 above)
Comment (shortcuts) [reply scripts/specs/format.ts:78]: it is only for clients build, done in `bundleSpecsForClient`

File: scripts/specs/format.ts (hunk 4 above)
Comment (shortcuts) [reply scripts/specs/format.ts:145]: yup this is still in place!

File: specs/bundled/ingestion.doc.yml
[the remark this answers is not in the document]
Comment (shortcuts) [reply specs/bundled/ingestion.doc.yml:1]: sure

File: scripts/specs/index.ts (hunk 2 above)
Comment (shortcuts) [reply scripts/specs/index.ts:113]: really not a fan of this syntax either D:

Comment (shortcuts) [reply scripts/specs/index.ts:113]: I can do a real if if you want

Comment (millotp) [reply scripts/specs/index.ts:113]: idk they both look bad

Comment (shortcuts) [reply scripts/specs/index.ts:113]: exactly ahahah

File: scripts/specs/index.ts (hunk 1 above)
Comment (shortcuts) [reply scripts/specs/index.ts:78]: ah good point I forgot to add it back

**Review by millotp — APPROVED** [review]
gg

**Pull request merged**